### PR TITLE
Mainframes: or why the fuck did I write a MOS 6502 emulator in BYOND.

### PR DESCRIPTION
**Pull request by TheChosenEvilOne**
## About The Pull Request
Adds a mainframe computer running a MOS 6502 emulator, for people to play around with.
image

TODO:
- [x] ~~Mapping (somewhere in science)~~ Mapped in on Meta and Delta.
- [x] Add main unit UI (reset, power, registers)
- [x] Add ROM unit UI (buttons for inserting and taking ROM banks + address information)
- [x] ROM bank editor sprite + UI improvements (?)
- [x] Add wiring to the main unit (interrupts, power)
- [x] Reorganise the types and files a bit.
- [x] Make it run at the intended speed (100 instructions per second).
- [x] Add persistent drawer for ROM banks so people don't always have to use the horrible ROM bank editor.
- [x] ~~Ask @Sparkezel to make a few more needed sprites (ROM bank editor, cassette item).~~ I made them myself
- [x] Peripheral linker UI.
- [x] ~~More peripherals (what?)~~ Added signaler, probably good enough for now.
- [x] ~~Write a guide explaining the memory map, etc. so people don't have to look at the code.~~ Added a quick reference because I am bad at explaining things.

I hate tgui

## Why It's Good For The Game
Dunno, I thought it would be funny. I guess you can get someone to program a text adventure game, also in the future it can be expanded to do all kinds of things.

## Changelog
:cl:
add: Added the mainframe, the mainframe is a programmable computer using a MOS 6502 emulator.
/:cl:

### CODE REVIEW

File: code/_compile_options.dm [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#define LOWMEMORYMODE
```
Comment (TheChosenEvilOne) [inline code/_compile_options.dm:1]: just a reminder for me to remove this before undrafting.

**Comment by TheChosenEvilOne** [comment]
okay, this is ready for reviews. I still need to write the guide and map it in though.

**Comment by TheChosenEvilOne** [comment]
I have come to the conclusion that I am incapable of writing a good guide for this.

**Comment by EdgeLordExe** [comment]
LGTM

**Pull request merged**